app/models/katello/erratum.rb

```diff
@@ -52,8 +52,9 @@ def self.applicable_to_hosts(hosts)
     end
 
     def self.applicable_to_hosts_dashboard(hosts)
+      ids = hosts.to_sql.sub(/SELECT.*FROM/, "SELECT #{::Host.connection.quote_table_name('hosts')}.#{::Host.connection.quote_column_name('id')} FROM")
```

Comment:
if there is a large amount of hosts this leads to very slow queries, as first we pluck a large amount of ids and then pass it into an in clause in a second query.
This way the SQL engine can optimize the internal query, as well as saving some memory needed to store the temporary array.